docs/metrics/tp-fp-fn-tn.md

```diff
@@ -0,0 +1,194 @@
+
+#### Single-class
+
+Let’s assume that a matching algorithm has already been run on all predictions and that the matched pairs and unmatched
+ground truths and predictions are given. Consider the following variables, adapted from
+[`match_inferences`][kolena.workflow.metrics.match_inferences]:
+
+| Variable | Type | Description |
+| --- | --- | --- |
+| `matched` | <nobr>`List[Tuple[GT, Inf]]`</nobr> | List of **matched** ground truth and prediction bounding box **pairs** (from Geometry Matcher) |
+| `unmatched_gt` | `List[GT]` | List of **unmatched** ground truth bounding boxes (from Geometry Matcher) |
+| <nobr>`unmatched_inf`</nobr> | `List[Inf]` | List of **unmatched** inference bounding boxes (from Geometry Matcher) |
+| `T` | `float` | Threshold used to filter valid prediction bounding boxes based on their confidence scores |
+
+Then these metrics are defined:
+
+```python
+TP = len([inf.score >= T for _, inf in matched])
+FN = len([inf.score <  T for _, inf in matched]) + len(unmatched_gt)
+FP = len([inf.score >= T for inf in unmatched_inf])
+```
+
+??? example "Example: Single-class Object Detection"
+
+    ![Legends](../assets/images/metrics-tpfpfntn-legends.png)
+
+    ![Single-class example](../assets/images/metrics-tpfpfntn-single-class.png)
+
+    This example includes two ground truths and two inferences, and when computed with an IoU threshold of 0.5 and
+    confidence score threshold of 0.5 yields:
+
+    | TP | FP | FN |
+    | --- | --- | --- |
+    | 1 | 1 | 1 |
+
+#### Multiclass
+
+Like classification, multiclass object detection tasks compute TP / FP / FN per class.
+
+??? example "Example: Multiclass Object Detection"
+
+    ![Legends](../assets/images/metrics-tpfpfntn-legends.png)
+
+    ![Multi-class example](../assets/images/metrics-tpfpfntn-multi-class.png)
+
+    Similar to multiclass classification, TP / FP / FN are computed for class `Apple` and class `Banana` separately.
+
+    Using an IoU threshold of 0.5 and a confidence score threshold of 0.5, this example yields:
+
+    | Class | TP | FP | FN |
+    | --- | --- | --- | --- |
+    | `Apple` | 0 | 0 | 1 |
+    | `Banana` | 0 | 1 | 0 |
+
+### Averaging Per-class Metrics
+
+For problems with multiple classes, these TP / FP / FN / TN metrics are computed for each class. If you are looking for
+a single score that summarizes model performance across all classes, there are a few different ways to aggregate
+per-class metrics: **macro**, **micro**, and **weighted.**
+
+Read more about these different averaging methods in the [Averaging Methods guide](./averaging-methods.md).
+
+## Limitations and Biases
+
+TP, FP, TN, and FN are four metrics based on the assumption that each sample/instance can be classified as a positive
+or a negative, thus they can only be applied to single-class applications. The workaround for multiple-class
+applications is to compute these metrics for each label using the **one-versus-rest** (OvR) strategy and then treat it
```

Comment:
"**one-vs-rest** (OvR)" used [here](https://github.com/kolenaIO/kolena/pull/91/files#diff-62887e7f8c765524b6957bd5f04cd5ed5328015562d7b14db39e5186872678deR81)

Should we use "vs" or "versus"?